routes.jsx

```diff
@@ -15,21 +16,38 @@ import Navbar from './components/navbar/navbar.jsx';
 import Footer from './components/footer/footer.jsx';
 
 const App = React.createClass({
+  getInitialState() {
+    return {
+      suppressSplashScreen: localstorage.getItem(`suppressSplashScreen`) === `true`
+    };
+  },
   componentDidMount() {
-    setTimeout(function() {
-      this.splash.classList.add(`dismissed`);
-      document.querySelector(`#app`).classList.add(`splash-dismissed`);
-    }, 3000);
+    if (this.state.suppressSplashScreen !== `true`) {
+      setTimeout(function() {
+        this.splash.classList.add(`dismissed`);
+        document.querySelector(`#app`).classList.add(`splash-dismissed`);
+      }, 3000);
+      localstorage.setItem(`suppressSplashScreen`, `true`);
+    }
+  },
+  renderWelcomeSplash() {
+    if (this.state.suppressSplashScreen) {
+      return null;
+    }
+    return (
+      <div id="splash" ref={(splash) => { this.splash = splash; }}>
```

Comment:
I don't understand this line. Why the need to assign this to an instance variable instead of just calling it `ref="splash"` and then accessing it using `this.refs.splash`? That's what refs are for. As you're only referencing it after mounting, there should be no problems relying on the normal ref behaviour.